helpers/post_proc_gard.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+import xarray as xr
+import numpy as np
+from scipy import stats
+
+varname = "PRECT"
+output_base = "output/gard_out_"
+
+has_pop = True
+
+mean_file = output_base+varname+".nc"
+err_file = output_base+varname+"_errors.nc"
+pop_file = output_base+varname+"_logistic.nc"
+
+# scrfs_file = "conus_scrfs_150yrs.nc"
+scrfs_file = "AR_scrfs.nc"
+output_file = "processed_{}.nc".format(varname)
+
+
+def main():
+
+    # open the primary data files
+    mean = xr.open_dataset(mean_file)
+    err = xr.open_dataset(err_file)
+    scrfs = xr.open_dataset(scrfs_file)
+
+    # apply the transform that was used within GARD
+    d1 = np.cbrt(mean[varname])
+
+    # get the random variables into a format that is appropriate in case the shapes don't match
+    rand_tmp = scrfs.p_rand_uniform[:d1.shape[0], :d1.shape[1], :d1.shape[2]]
+
+    # if there is a probability of precipitation (PoP) or other threshold value specified
+    # then we need to do some extra work to use that prediction
+    if has_pop:
+        # load the PoP dataset
+        pop = xr.open_dataset(pop_file)
+        popval = pop[varname+"_exceedence_probability"]
+
+        # figure out where the probability was not exceeded
+        noprecip = rand_tmp.values < (1-popval.values)
+        # rescale the random numbers to account for the limited portion of the cdf they should occupy
+        rand_tmp.values -= (1-popval.values)
+        rand_tmp /= popval.values
+
+    # convert uniform random numbers to normally distributed values
+    rand_tmp.values[rand_tmp.values<=1e-20] = 1e-20
+    errors = stats.norm.ppf(rand_tmp.values)
```

Comment:
Ideally, we don't have to pull out the values here. I think you can stick with xarray/dask via something like this:

```Python
LOWER_THRESH = 1e-20
rand_tmp = rand_tmp.where(rand_tmp < LOWER_THRESH, LOWER_THRESH)
errors = ppf(rand_tmp)
```

You probably want to use the xarray ufunc version of ppf and cbrt available here: 

https://github.com/jhamman/storylines/blob/8bf2869ec665f9472adb6c9213b9b570bf85e8bf/storylines/tools/post_process_gard_output.py#L36-L45

Reply:
Yes, if you have the time to simplify your script to remove the aspects related to, e.g., RCP dates and other storylines work, that would be great.  Unless you can do that today / tomorrow, then I’d say let’s merge this in, but plan on replacing it as soon as feasible with yours.  Maybe setting yours up to read the gard configuration file so it automatically uses the correct files / options.  Keep in mind that GARD output files are not CF compliant / xarray friendly... scrf files may not be.

Reply:
(That should have been a reply on the larger review comment, not this particular comment)